### DISCUSSION TITLE
Webhook telegram how to make dld of now playing song possible ?

### DISCUSSION THREAD

**Discussion by Schrebergarten03-links** [AzuraCast/AzuraCast#5512] [Q&A]
Hi im using

Title: {{ now_playing.song.title }}
Artist: {{ now_playing.song.artist }} Album: {{ now_playing.song.album }}
Playlist: {{ now_playing.playlist }} Listeners: {{listeners.unique}}
Web: {{ station.public_player_url }}

in a telegram webhook chat bot ... workes nice and solid but i would like to add a on demand download link to the now playing song.

that should look like

https:// url/api/station/xy/ondemand/download/cdb99bb4a7cc07ea97xxxxxx

but for that i need the :

Api_StationOnDemand{ track_id string } (Track ID unique identifier) 25-character

but there is only a

Api_NowPlaying_CurrentSong{ song Api_Song{ id string example: 9f33bbc912c19603e51be8e098xxxxxx (32-character unique identifier hash)

Long story short id like to make the audience able to easyly dld a song they like via telegram webhook...
maybe i didnt get it right im new to the whole thing ^^and there is already a possibility ... then it would be nice if anyone can show me how

greets

Ps:

Also it would be nice if the webhook function had options to change: "All values in the NowPlaying API response are available for use." to a checkbox list of all api options so one can maybe remote control a radio via telgram bot comands and or do wattever other stuff.
